### PR TITLE
WIP: Add `electric_plant_depreciation_changes_ferc1` into explosion

### PR DESCRIPTION
**Pull request by e-belfer**
A sub-task of #2016. At present, we are missing `electric_plant_depreciation_changes_ferc1`, which is a sub-component of `balance_sheet_assets_ferc1`. This PR does the following:
- subclasses `process_xbrl_metadata()` to account for a non-standard renaming of the XBRL factoid provided in the FERC Instant XBRL data
- renames `utility_plant_value` to `dollar_value` to be consistent with other tables
- adds calculation and second-dimension calculation checks into the table processing
- adds one calculation into the metadata which is missing (`ending_balance`)

To do:
- [ ] Right now ~35% of calculations are off with no very obvious candidate. Fix this!

-->

# PR Overview

# PR Checklist

- [ ] Merge the most recent version of the branch you are merging into (probably `dev`).
- [ ] All CI checks are passing. [Run tests locally to debug failures](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
- [ ] Make sure you've included good docstrings.
- [ ] For major data coverage & analysis changes, [run data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation)
- [ ] Include unit tests for new functions and classes.
- [ ] Defensive data quality/sanity checks in analyses & data processing functions.
- [ ] Update the [release notes](https://catalystcoop-pudl.readthedocs.io/en/latest/release_notes.html) and reference reference the PR and related issues.
- [ ] Do your own explanatory review of the PR to help the reviewer understand what's going on and identify issues preemptively.

### CODE REVIEW

File: src/pudl/transform/ferc1.py [hunk 1]
```diff
@@ -1316,7 +1312,7 @@ def process_xbrl_metadata(self: Self, xbrl_metadata_json) -> pd.DataFrame:
         tbl_meta.xbrl_factoid = tbl_meta.xbrl_factoid.map(xbrl_factoid_name_map)
 
         def rename_calculation_components(calc: str) -> str:
-            # apply the rename for the "name" element of all of the calc components
+            """Apply the rename for the "name" element of all of the calc components."""
```
Comment (e-belfer) [inline src/pudl/transform/ferc1.py:1315]: fossil from another code change that can be reverted

File: src/pudl/transform/ferc1.py [hunk 2]
```diff
@@ -4731,16 +4770,24 @@ class ElectricPlantDepreciationChangesFerc1TableTransformer(
...
-        Replace the name of the balance column reported in the XBRL Instant table with
-        starting_balance / ending_balance since we pull those two values into their own
-        separate labeled rows, each of which should get the original metadata for the
-        Instant column.
+        Warning: The calculations in this table are currently being checked using
+        reconcile_table_calculations(), but they still contain high rates of error.
+        This function replaces the name of the balance column reported in the XBRL
+        Instant table with starting_balance / ending_balance since we pull those two
+        values into their own separate labeled rows, each of which should get the
+        original metadata for the Instant column. We do this pre-processing before we
```
Comment (aesharpe) [inline src/pudl/transform/ferc1.py:4778]: This bit is a little hard to follow. Might be that I don't understand the context of the table, though.

File: src/pudl/transform/ferc1.py [hunk 3]
```diff
@@ -4731,16 +4770,24 @@ class ElectricPlantDepreciationChangesFerc1TableTransformer(
...
         """
-        meta = super().process_xbrl_metadata(xbrl_metadata_json)
-        ending_balance = meta[meta.xbrl_factoid == "starting_balance"].assign(
-            xbrl_factoid="ending_balance"
+        new_xbrl_metadata_json = xbrl_metadata_json
+        instant = pd.json_normalize(xbrl_metadata_json["instant"])
+        instant = pd.concat([instant] * 2).reset_index(drop=True)
+        instant["name"] = instant["name"] + ["_starting_balance", "_ending_balance"]
+        new_xbrl_metadata_json["instant"] = json.loads(
+            instant.to_json(orient="records")
```
Comment (aesharpe) [inline src/pudl/transform/ferc1.py:4787]: Can you add some comments here to explain what is happening?

File: src/pudl/transform/ferc1.py [hunk 4]
```diff
@@ -1311,7 +1311,7 @@ def process_xbrl_metadata(self: Self, xbrl_metadata_json) -> pd.DataFrame:
         tbl_meta.xbrl_factoid = tbl_meta.xbrl_factoid.map(xbrl_factoid_name_map)
 
         def rename_calculation_components(calc: str) -> str:
-            # apply the rename for the "name" element of all of the calc components
+            # Apply the rename for the "name" element of all of the calc components.
```
Comment (aesharpe) [inline src/pudl/transform/ferc1.py:1314]: This description isn't super clear.

File: src/pudl/transform/ferc1.py [hunk 5]
```diff
@@ -1962,6 +1962,45 @@ def apply_xbrl_calculation_fixes(
                     },
                 ],
             },
+            "electric_plant_depreciation_changes_ferc1": {
+                "ending_balance": [
+                    {
+                        "calc_component_to_replace": {},
+                        "calc_component_new": {
+                            "name": "starting_balance",
+                            "weight": 1.0,
```
Comment (aesharpe) [inline src/pudl/transform/ferc1.py:1971]: How did you determine that these were the calculation components of `ending_balance`? Did you run the calculation and see that it was mostly accurate? Is it based on the components used in other tables? Would be good to document this somewhere.

**Review by aesharpe — APPROVED** [review]
Looks good to me! Nothing super blocking, just requests for clarification.
